examples/src/bin/orientable_subclass.rs

```diff
@@ -0,0 +1,176 @@
+use std::cell::RefCell;
+use std::env;
+
+use gtk::glib;
+use gtk::prelude::*;
+use gtk::subclass::prelude::ObjectSubclass;
+
+mod imp {
+    use super::*;
+    use gtk::{glib::translate::ToGlib, subclass::prelude::*};
+
+    #[derive(Debug)]
+    pub struct CustomOrientable {
+        first_label: RefCell<Option<gtk::Widget>>,
+        second_label: RefCell<Option<gtk::Widget>>,
+        orientation: RefCell<gtk::Orientation>,
+    }
+
+    // Every widget that implements Orientable has to define a "orientation"
+    // property like below, gtk::Orientation::Horizontal is a placeholder
+    // for the initial value.
+    //
+    // glib::ParamFlags::CONSTRUCT allows us to set that property the moment
+    // we create a new instance of the widget
```

Comment:
That's always possible, but `CONSTRUCT` properties are set before all the others. It rarely makes a difference.

But `CONSTRUCT` means that it will even be passed to `set_property()` on construction if it was not specified as part of `glib::Object::new()`. In that case the default property value would be used.

Reply:
The thing is that if you set Vertical everywhere, without that flag, it would start as Horizontal anyways. The setter only sends the value to the layout manager on the first set.

Reply:
I don't understand what you mean, sorry. Can you explain in more detail what happens and what problem you observe?

Reply:
We are currently setting horizontal as the default value of the PROPERTY, and in the RefCell in new(). I don't know if this is a problem, but if you switch them both to vertical, the initial (visible) state will still be horizontal, my guess is that the setter (which changes the layout manager) has not kicked yet.

Reply:
@bilelmoussaoui is that how layout managers work?

From the Rust side, you should get things set correctly at least.

Reply:
Layout managers are currently a bit weird to use. And the current example is trying to implement a GtkBox without really implementing one. I would say it's okay as is, and yes the property has to be set at least once, so that the layout manager's property here gets updated. That's why passing a value when we construct the widget is required in such case

Reply:
You don't have to provide it to the `new()` call if you use the `CONSTRUCT` flag for the property.

Reply:
Ah; TIL

Reply:
So how should we proceed here? :)

Reply:
Works for me